group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -8179,29 +7770,30 @@ public void testGenerationIdIncrementsOnRebalance() throws Exception {
             .build();
         JoinGroupResponseData leaderJoinResponse = context.joinGenericGroupAsDynamicMemberAndCompleteRebalance("group-id");
 
-        CompletableFuture<JoinGroupResponseData> joinFuture = new CompletableFuture<>();
-        CoordinatorResult<Void, Record> result = context.sendGenericGroupJoin(
+        JoinResult joinResult = context.sendGenericGroupJoin(
             new JoinGroupRequestBuilder()
                 .withGroupId("group-id")
                 .withMemberId(leaderJoinResponse.memberId())
                 .withDefaultProtocolTypeAndProtocols()
-                .build(),
-            joinFuture);
+                .build()
+        );
 
-        assertTrue(result.records().isEmpty());
-        assertTrue(joinFuture.isDone());
-        assertEquals(2, joinFuture.get().generationId());
-        assertEquals(Errors.NONE.code(), joinFuture.get().errorCode());
+        assertTrue(joinResult.records.isEmpty());
+        assertTrue(joinResult.joinFuture.isDone());
+        assertEquals(2, joinResult.joinFuture.get().generationId());
+        assertEquals(Errors.NONE.code(), joinResult.joinFuture.get().errorCode());
     }
 
     private List<JoinGroupResponseMember> toJoinResponseMembers(GenericGroup group) {
```

Comment:
nit: This method could be static? checkJoinGroupResponse as well. Are those two helpers as well?